client/client-lib/src/wallet/mod.rs

```diff
@@ -98,27 +98,35 @@ impl WalletClient {
         address
     }
 
-    pub fn create_pegin_input(
+    pub async fn create_pegin_input(
         &self,
         txout_proof: TxOutProof,
         btc_transaction: bitcoin::Transaction,
     ) -> Result<(KeyPair, PegInProof)> {
-        let (output_idx, secret_tweak_key_bytes) = btc_transaction
-            .output
-            .iter()
-            .enumerate()
-            .find_map(|(idx, out)| {
+        let output_function = || async {
+            for (idx, out) in btc_transaction.output.iter().enumerate() {
                 debug!(output_script = ?out.script_pubkey);
-                self.context
+                let result = self
+                    .context
                     .db
                     .begin_transaction(ModuleRegistry::default())
                     .get_value(&PegInKey {
                         peg_in_script: out.script_pubkey.clone(),
                     })
+                    .await
                     .expect("DB error")
-                    .map(|tweak_secret| (idx, tweak_secret))
-            })
+                    .map(|tweak_secret| (idx, tweak_secret));
```

Comment:
Not sure if this is concurrency-safe, but also not sure if we should aim for that with the legacy client lib.